Bluetooth: samples: Peripheral and Central UART fix buffer releasing (#7190)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

    * Increased the main stack size from 1024 to 1536 bytes.

  * :ref:`peripheral_uart` sample:

>     * Fixed double attempt to free the UART buffer.

Reviewer: I think we could elaborate this a bit. Is the double attempt something that was in the sample and it is not there anymore? Not sure how to interpret this.

Author: Ok. I wrote the longer paragraph about my changes :)

    * Increased the main stack size from 1024 to 1536 bytes.

  * :ref:`peripheral_uart` sample:

>     * Fixed double attempt to free the Rx UART buffer.
>       The Rx buffer might have been released twice by main thread.
>       Handling Rx buffer releasing is now fixed in this sample and in the UARTE driver.

Reviewer: ```suggestion
    * Fixed handling of Rx buffer releasing in this sample and in the UARTE driver.
      Before this fix, the Rx buffer might have been released twice by the main thread.
```
How about this?

Author: I like your proposal. Done

  * :ref:`central_uart` sample:

>     * Fixed double attempt to free the Rx UART buffer.
>       The Rx buffer might have been released twice by main thread.
>       Handling Rx buffer releasing is now fixed in this sample and in the UARTE driver.

Reviewer: ```suggestion
    * Fixed handling of Rx buffer releasing in this sample and in the UARTE driver.
      Before this fix, the Rx buffer might have been released twice by the main thread.
```

Author: Done
